### PR TITLE
Fix inconsistency between T5WithLMHeadModel's doc and it's behavior

### PR DESCRIPTION
**Pull request by nalourie-ai2**
The doc string for `T5WithLMHeadModel` is currently inconsistent with it's behavior.

The doc string says that the forward method ignores indices of -1 when computing the loss; however, the method instead ignores indices of -100. This pull request changes the method to ignore indices of -1, making the two consistent.

It's worth noting, there is a [commit](https://github.com/huggingface/transformers/commit/1b59b57b57010e6119282f3dbf37f8c7c6d6313e#diff-7370db3a19209bf984cc40925aaf2b71) by @thomwolf that changed this value from -1 to -100, though I couldn't find why the change was made. Thomas, perhaps you remember if the change is still important? If it is, I can instead update this PR to change the doc string.

### CODE REVIEW

**Comment by LysandreJik** [comment]
Hi! Indeed, we changed all the Cross Entropy ignored indices to be -100 to respect the PyTorch default. The docstrings would need to be changed, instead of the index. We could even remove the argument:

```py
loss_fct = CrossEntropyLoss()
```

to keep in sync with pytorch.

**Comment by nalourie-ai2** [comment]
@LysandreJik, that makes perfect sense. Thanks for the background!

I've updated this PR to change the doc string instead. 

Removing the `ignore_index` argument would mean that the behavior of the method depends on which PyTorch version a user has installed, which could be counter-intuitive. Similarly, to ensure the doc string is accurate we'd have to add a unit test for it and run against all versions of PyTorch that a user might reasonably install.

For simplicity / consistency, I'd suggest continuing to explicitly pass the argument.

I'm happy to add a commit removing it though, if you feel otherwise.

**Comment by LysandreJik** [comment]
Thanks for updating, we can keep the `ignore_index` argument.

Thanks @nalourie-ai2 !
